BlockchainSdk/WalletManagers/Chia/ChiaTransactionBuilder.swift

```diff
@@ -0,0 +1,162 @@
+    /// - Returns: Sum value for transaction
+    func getTransactionCost(amount: Amount) -> UInt64 {
+        let decimalAmount = amount.value / blockchain.decimalValue
+        let decimalBalance = unspentCoins.map { Decimal($0.amount) }.reduce(0, +)
+        let change = decimalBalance - decimalAmount
+        let numberOfCoinsCreated: Int = change > 0 ? 2 : 1
+
+        return UInt64((coinSpends.count * CostConstants.COIN_SPEND_COST) + (numberOfCoinsCreated * CostConstants.CREATE_COIN_COST))
+    }
+    
+    // MARK: - Private Implementation
+    
+    private func calculateChange(transaction: Transaction, unspentCoins: [ChiaCoin]) throws -> UInt64 {
+        let fullAmount = unspentCoins.map { $0.amount }.reduce(0, +)
+        let transactionAmount = transaction.amount.value * blockchain.decimalValue
+        let transactionFeeAmount = transaction.fee.amount.value * blockchain.decimalValue
+        let changeAmount = fullAmount - (transactionAmount.uint64Value + transactionFeeAmount.uint64Value)
+        
+        return changeAmount
+    }
+    
+    private func toChiaCoinSpends(change: UInt64, destination: String, source: String, amount: Amount) throws -> [ChiaCoinSpend] {
+        var coinSpends = unspentCoins.map {
+            ChiaCoinSpend(
+                coin: $0,
+                puzzleReveal: ChiaPuzzleUtils().getPuzzleHash(from: walletPublicKey).hex,
+                solution: ""
+            )
+        }
+        
+        let sendCondition = try createCoinCondition(for: destination, with: amount.value.uint64Value)
+        let changeCondition = try change != 0 ? createCoinCondition(for: source, with: change) : nil
+        
+        let solution: [ChiaCondition] = [sendCondition, changeCondition].compactMap { $0 }
+        coinSpends[0].solution = try solution.toSolution().hex
+        
+        for var coinSpend in coinSpends.dropFirst(1) {
+            coinSpend.solution = try [RemarkCondition()].toSolution().hex
+        }
+
+        return coinSpends
+    }
+    
+    private func createCoinCondition(for address: String, with change: UInt64) throws -> CreateCoinCondition {
+        return try CreateCoinCondition(
+            destinationPuzzleHash: ChiaPuzzleUtils().getPuzzleHash(from: address),
+            amount: change
+        )
+    }
+    
+}
+
+// MARK: - Constants
+
+fileprivate extension ChiaTransactionBuilder {
+    enum CostConstants {
+        static let COIN_SPEND_COST: Int = 4500000
+        static let CREATE_COIN_COST: Int = 2400000
+    }
+    
+    enum ChiaGenesisChallenge {
+        private static let genesisChallengeMainnet = "ccd5bb71183532bff220ba46c268991a3ff07eb358e8255a65c30a2dce0e5fbb"
+        private static let genesisChallengeTestnet = "ae83525ba8d1dd3f09b277de18ca3e43fc0af20d20c4b3e92ef2a48bd291ccb2"
+        
+        static func genesisChallenge(isTestnet: Bool) -> String {
+            return isTestnet ? genesisChallengeTestnet : genesisChallengeMainnet
+        }
+    }
+}
```

Comment:
```suggestion
    enum GenesisChallenge {
        private static let mainnet = "ccd5bb71183532bff220ba46c268991a3ff07eb358e8255a65c30a2dce0e5fbb"
        private static let testnet = "ae83525ba8d1dd3f09b277de18ca3e43fc0af20d20c4b3e92ef2a48bd291ccb2"
        
        static func challenge(isTestnet: Bool) -> String {
            return isTestnet ? testnet : mainnet
        }
    }
}
```
по факту в названии же есть `genesisChallenge` можно и не писать

Reply:
В целом да, но делал также чтобы с Андроидом совпадало. Твой вариант выглядит симпатичней